Source/NSView.m

```diff
@@ -5212,6 +5217,35 @@ - (void) _setNeedsUpdateConstraints: (BOOL)needsUpdateConstraints
   _needsUpdateConstraints = needsUpdateConstraints;
 }
 
+- (void) _layoutViewAndSubViews
+{
+  if (_needsLayout)
+    {
+      [self layout];
+      _needsLayout = NO;
+    }
+
+  NSArray *subviews = [self subviews];
+  FOR_IN (NSView *, subview, subviews)
+    [subview _layoutViewAndSubViews];
+  END_FOR_IN (subviews);
+}
+
+- (GSAutoLayoutEngine*) _layoutEngine
+{  
+  return _layoutEngine;
+}
+
+- (void) _setLayoutEngine: (GSAutoLayoutEngine *)engine
+{
+  _layoutEngine = engine;
```

Comment:
The reference handling for the layout engine is not clear to me. Who is responsible to retain and free it? The current implementation could leak it as well as release it multiple times. When a view gets removed from its superview and they have a shared layout engine.

Reply:
My intention was for it to be owned by the top most view of the view hierarchy. As the view hierarchy can be modified this introduces a lot of complexity. I've move the ownership of the layout engine to NSWindow, this simplifies things. However I am under the impression that AutoLayout works with a view without a window, so I might need to revisit this at a later stage.